### PR TITLE
Add badges

### PR DESCRIPTION
**Pull request by Jeff-Thompson12**
Add badges to README

### CODE REVIEW

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK The main thing this PR is missing is a Github Action to automate the rendering of README.Rmd. Based on our conversation yesterday, I think we are wanting to do that automation in a separate PR. 

I mimicked the creation of the metrics found in the [pharmaverse](https://pharmaverse.org/e2eclinical/tlg/). We can obviously change both our calculations and how we want to shade the badges.

**Comment by Jeff-Thompson12** [comment]
Addresses #92 and #74

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
Hi @Jeff-Thompson12, Thanks for the PR! However...

(1) It did add a new `note` to the [R CMD Check](https://github.com/Biogen-Inc/tidyCDISC/actions/runs/3093019953/jobs/5004925444) regarding a non-standard folder name in the root dir:

image

Consider adding the scripts to `~/inst/app/www/scripts/` as an alternative or maybe `~/man/scripts/`. Whichever makes the most sense. I think the `README.Rmd` already leverages a lot of figures stored in `~/man/` so maybe that's the best option?

(2) `{GithubMetrics}` and `{riskmetric}` should probably be added to the Suggests field using:

```
usethis::use_dev_package("openpharma/GithubMetrics", type = "Suggests")
usethis::use_package("riskmetric", type = "Suggests")
```

(3) shouldn't `{renv}` download `{GithubMetrics}` for me? I had to manually install it. Maybe I need to run some `{renv}` command to update. Sorry, I am still an `{renv}` noob.

(4) In general (and only if possible), can you include a small screenshot of the proposed changes in your PR? I got this result when rendering the README (below). Is that what you got? 

image

The riskmetric score seems... high, lol. I didn't know it was even possible to get a 1.0! I ran tidyCDISC through the [Risk Assessment App (beta version) ]([rinpharma.shinyapps.io/risk_assessment](https://rinpharma.shinyapps.io/risk_assessment)) and got the following score of 0.45:

image

Thanks!

File: scripts/riskmetric.R [hunk 1]
```diff
@@ -0,0 +1,47 @@
+library(glue)
+library(dplyr)
+library(riskmetric)
+
+riskmetric_score <- "tidyCDiSC" %>%
+  pkg_ref() %>%
+  as_tibble() %>%
+  pkg_assess() %>%
+  pkg_score() %>%
+  pull("pkg_score")
```
Comment (AARON-CLARK) [inline scripts/riskmetric.R:10]: Hi @Jeff-Thompson12, here are the details of running this code locally:

image

**Comment by Jeff-Thompson12** [comment]
> (3) shouldn't `{renv}` download `{GithubMetrics}` for me? I had to manually install it. Maybe I need to run some `{renv}` command to update. Sorry, I am still an `{renv}` noob.

When you switch to the add_badges branch, (1) restart R (Ctrl + Shift + F10) then (2) run `renv::status()` which will show you which of your packages are out of synch, you can then (3) run `renv::restore()` to set your library to the lockfile.

So you still have to run code, but it should install it for you when running `renv::restore()`.

**Comment by Jeff-Thompson12** [comment]
Regarding (1)

> (1) It did add a new `note` to the [R CMD Check](https://github.com/Biogen-Inc/tidyCDISC/actions/runs/3093019953/jobs/5004925444) regarding a non-standard folder name in the root dir:
> 
> image
> 
> Consider adding the scripts to `~/inst/app/www/scripts/` as an alternative or maybe `~/man/scripts/`. Whichever makes the most sense. I think the `README.Rmd` already leverages a lot of figures stored in `~/man/` so maybe that's the best option?
> 
> (2) `{GithubMetrics}` and `{riskmetric}` should probably be added to the Suggests field using:
> 
> ```
> usethis::use_dev_package("openpharma/GithubMetrics", type = "Suggests")
> usethis::use_package("riskmetric", type = "Suggests")
> ```

Regarding (1) and (2), I see the scripts as outside of the R package itself. They exist merely to calculate values. The README.md file is static. I do not think they should be exported in an R build.

**Comment by AARON-CLARK** [comment]
> I do not think they should be exported in an R build.

I still think (2) is valid. Packages in the `Suggests` field won't be installed with the build, but they are just nice to have for other purposes in development, as Karl Broman wrote in his [R-pkgs book](https://kbroman.org/pkg_primer/pages/depends.html). To my knowledge, there is no penalty for adding them, correct? So why not add them?

image

**Comment by AARON-CLARK** [comment]
Friendly reminder to post a screenshot of your results! I'm getting something outside the [0, 1] bounds: I just realized you never did re-request my review on this PR, so maybe it's not ready yet? Sorry!

image

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK What version of {riskmetric} are you using?

**Comment by AARON-CLARK** [comment]
> @AARON-CLARK What version of {riskmetric} are you using?


image

Same as `renv.lock`:
image

**Comment by AARON-CLARK** [comment]
Ohhh, I just realized I don't have the CRAN version of tidyCDISC installed. It would help if I ran my `renv::restore`

image

**Comment by AARON-CLARK** [comment]
I re-ran with `tidyCDISC 0.1.0` installed and my score is still negative. Can I ask what score you are getting? The same as the Risk Assessment App?

image

**Comment by Jeff-Thompson12** [comment]
I am not getting the exact same value as the risk assessment app. The value is in the README.md file.
image

Your badge looks funny to me. I'm trying to figure out why it would even look like that. It looks more similar to [visR ](https://github.com/openpharma/visR#visr-) than what I thought I created.

**Comment by Jeff-Thompson12** [comment]
Also, thought that just occurred to me, I have the branch loaded when I run the code. I need to restart my R session before running it.

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK since {GithubMetrics} is not available on CRAN, it needed to be added in the `Remotes:`. We would not be able to resubmit to CRAN with a `Remotes:` section, so I do not know if that makes us more inclined not to include the packages in `Suggests:` or `Remotes:`. Let me know your thoughts.

**Comment by AARON-CLARK** [comment]
Aaron is getting a -.3, but that's because he has tidyCDISC installed.

**Comment by AARON-CLARK** [comment]
@Jeff-Thompson12: so, I did a little digging and found the following:

```
library(dplyr)
library(riskmetric)

"tidyCDISC" %>%
  pkg_ref() %>%
  as_tibble() %>%
  pkg_assess() %>%
  pkg_score(error_handler = function(x) {return(0)}) %>%
  pull("pkg_score") %>%
  round(2) 

> .51
```
However, the app is producing:
image

At first, I thought the differing `pkg_score` might have something to do with the app's `covr_coverage` weight being set to zero. So, I wanted to mimic that locally:
```
t <- 
"tidyCDISC" %>%
  pkg_ref() %>%
  as_tibble() %>%
  pkg_assess() %>%
  pkg_score(error_handler = function(x) {return(0)}) 

assess_names <- names(t)[5:19]
# assess_names
metric_weights <- c(rep(1,length(assess_names)-1), 0)
metric_weights <- setNames(metric_weights, assess_names)

"tidyCDISC" %>%
  pkg_ref() %>%
  as_tibble() %>%
  pkg_assess() %>%
  pkg_score(weights = metric_weights, 
            error_handler = function(x) {return(0)}) %>%
  pull("pkg_score") %>%
  round(2)

> 0.47
```

So, we are closer! But still far away from .42. Then I realized the problem, there are 3 "new" metrics being assessed by `riskmetric` that we aren't taking into consideration: `r_cmd_check`, `remote_checks`, and `exported_namespace`. I'll open up a new issue with `riskassessment` to add these, but I think we can assume our local `riskmetric` code is correct.

So, to solve a couple other issues, `GithubMetrics` package totally exists on CRAN, so I think we could switch over to using that instead of the latest dev version on Github, so I think you can use `usethis::use_package("GithubMetrics", type = "Suggests")` to remove any `remotes` section issues, and also update `renv`. For some reason, the CRAN version does produce a slightly lower score, which is unfortunate. I re-rendered the `README.Rmd` and got the following stats:

image

Let me know when you think it's ready to go!

**Comment by Jeff-Thompson12** [comment]
> So, to solve a couple other issues, `GithubMetrics` package totally exists on CRAN,

Am I dumb? Why can I not find it? I don't even see it in the archive.

**Comment by AARON-CLARK** [comment]
> > So, to solve a couple other issues, `GithubMetrics` package totally exists on CRAN,
> 
> Am I dumb? Why can I not find it? I don't even see it in the archive.

@Jeff-Thompson12, apparently I am the dumb one, lol. For some reason, they say it's available on CRAN on their [README ](https://github.com/openpharma/GithubMetrics#installation)and so I just used:

image

But I didn't notice, it found the package at api.github.com. Whoops

**Comment by Jeff-Thompson12** [comment]
> I re-ran with `tidyCDISC 0.1.0` installed and my score is still negative. Can I ask what score you are getting? The same as the Risk Assessment App?

@AARON-CLARK I encountered this when using `devtools::build_readme()` instead of knitting. We never did figure out what was messing up the score when `{tidyCDISC}` is installed in the environment. I do not think that is a problem for this PR though.

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK This PR is ready for another look on the review front.

**Pull request merged**